packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -422,6 +475,32 @@ - (BOOL)setCaptureSessionPreset:(FLTResolutionPreset)resolutionPreset withError:
   return YES;
 }
 
+/// Finds the highest available resolution in terms of pixel count for the given device
+- (AVCaptureDeviceFormat *)highestResolutionFormatForCaptureDevice:
+    (AVCaptureDevice *)captureDevice {
+  AVCaptureDeviceFormat *bestFormat = nil;
+  NSUInteger maxPixelCount = 0;
+  for (AVCaptureDeviceFormat *format in [_captureDevice formats]) {
+    CMVideoDimensions res = [self videoDimensionsForFormat:format];
+    NSUInteger height = res.height;
+    NSUInteger width = res.width;
+    NSUInteger pixelCount = height * width;
+    if (pixelCount > maxPixelCount) {
+      maxPixelCount = pixelCount;
+      bestFormat = format;
+    }
+  }
+  return bestFormat;
+}
+
+/// Wrapper for CMVideoFormatDescriptionGetDimensions to use custom code for testing
+- (CMVideoDimensions)videoDimensionsForFormat:(AVCaptureDeviceFormat *)format {
+  if (self.videoDimensionsForFormatBlock) {
+    return self.videoDimensionsForFormatBlock(format);
+  }
+  return CMVideoFormatDescriptionGetDimensions(format.formatDescription);
```

Comment:
This also goes in the inits, and this whole wrapper method goes away.

Reply:
fixed